test/scaladoc/run/SI-9620.scala

```diff
@@ -0,0 +1,43 @@
+import scala.tools.nsc.doc.model._
+import scala.tools.partest.ScaladocModelTest
+
+object Test extends ScaladocModelTest {
+  override def code = """
+    package a
+
+    trait Foo[S] {
+      def foo(t: S): Int = 123
+    }
+
+    /** Boo with only one foo method, hopefully!
+     * @hideImplicitConversion BooShouldNotAppearIsFoo
+     */
+    trait Boo[T]
+
+    object Boo {
+      sealed trait ShouldNotAppear
+      implicit class BooShouldNotAppearIsFoo(boo: Boo[ShouldNotAppear]) extends Foo[ShouldNotAppear]
+      implicit class BooLongIsFoo(boo: Boo[Long]) extends Foo[Long]
+    }
+  """
+
+  // no need for special settings
+  def scaladocSettings = ""
+
+  def testModel(rootPackage: Package) = {
+    import access._
+
+    // Assert Boo only has one implicit conversion
+    val boo = rootPackage._package("a")._trait("Boo")
+    val conversions = boo._conversions("a.Boo.BooShouldNotAppearIsFoo") ++ boo._conversions("a.Boo.BooLongIsFoo")
+    assert(conversions.length == 1, conversions.length + " == 1")
```

Comment:
Yep, most likely. Here's a test for implicits:
https://github.com/scala/scala/blob/2.12.x/test/scaladoc/run/implicits-known-type-classes.scala#L11